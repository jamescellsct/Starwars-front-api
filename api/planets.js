import express from "express";

const router = express.Router();

var planets = require("./fakeData/db.json").planets;

router.get("/planets", function(req, res) {
  return res.status(200).json({
    count: planets.count,
    results: planets.results[0]
  });
});

router.get("/planets/:page", function(req, res) {
  let { page } = req.params;

  page = page.slice(5, 6) - 1;

  if (!planets.results[page]) {
    return res.status(404).send({
      error: {
        message: "Page Not Found!"
      }
    });
  }

  return res.status(200).json({
    count: planets.count,
    results: planets.results[page]
  });
});

export default router;
