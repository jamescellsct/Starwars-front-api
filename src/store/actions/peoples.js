import { peoplesAction } from "./actionsTypes";
import { getPeople } from "../../services/api/peopleService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchPeopleStarted = () => ({
  type: peoplesAction.FETCH_PEOPLE_STARTED
});

const fetchPeopleSucceeded = people => ({
  type: peoplesAction.FETCH_PEOPLE_SUCCEEDED,
  people
});

const fetchPeopleFailed = error => ({
  type: peoplesAction.FETCH_PEOPLE_FAILED,
  error
});

export const fetchPeople = () => dispatch => {
  dispatch(fetchPeopleStarted());

  return getPeople()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchPeopleSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchPeopleFailed(error));
    });
};

const updatePeopleIsFetching = () => ({
  type: peoplesAction.UPDATE_PEOPLE_IS_FETCHING
});

const updatePeopleSucceeded = people => ({
  type: peoplesAction.UPDATE_PEOPLE_SUCCEEDED,
  people
});

const updatePeopleFailed = error => ({
  type: peoplesAction.UPDATE_PEOPLE_FAILED,
  error
});

export const updatePeople = page => dispatch => {
  dispatch(updatePeopleIsFetching());

  getPeople(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updatePeopleSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updatePeopleFailed(error));
    });
};
