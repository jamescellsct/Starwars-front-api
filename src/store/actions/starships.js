import { starshipsAction } from "./actionsTypes";
import { getStarships } from "../../services/api/starshipsService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchStarshipsStarted = () => ({
  type: starshipsAction.FETCH_STARSHIPS_STARTED
});

const fetchStarshipsSucceeded = starships => ({
  type: starshipsAction.FETCH_STARSHIPS_SUCCEEDED,
  starships
});

const fetchStarshipsFailed = error => ({
  type: starshipsAction.FETCH_STARSHIPS_FAILED,
  error
});

export const fetchStarships = () => dispatch => {
  dispatch(fetchStarshipsStarted());

  return getStarships()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchStarshipsSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchStarshipsFailed(error));
    });
};

const updateStarshipsIsFetching = () => ({
  type: starshipsAction.UPDATE_STARSHIPS_IS_FETCHING
});

const updateStarshipsSucceeded = starships => ({
  type: starshipsAction.UPDATE_STARSHIPS_SUCCEEDED,
  starships
});

const updateStarshipsFailed = error => ({
  type: starshipsAction.UPDATE_STARSHIPS_FAILED,
  error
});

export const updateStarships = page => dispatch => {
  dispatch(updateStarshipsIsFetching());
  return getStarships(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updateStarshipsSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updateStarshipsFailed(error));
    });
};
