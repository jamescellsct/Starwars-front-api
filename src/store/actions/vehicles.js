import { vehiclesAction } from "./actionsTypes";
import { getVehicles } from "../../services/api/vehiclesService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchVehiclesStarted = () => ({
  type: vehiclesAction.FETCH_VEHICLES_STARTED
});

const fetchVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
  vehicles
});

const fetchVehiclesFailed = error => ({
  type: vehiclesAction.FETCH_VEHICLES_FAILED,
  error
});

export const fetchVehicles = () => dispatch => {
  dispatch(fetchVehiclesStarted());

  return getVehicles()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchVehiclesSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchVehiclesFailed(error));
    });
};

const updateVehiclesIsFetching = () => ({
  type: vehiclesAction.UPDATE_VEHICLES_IS_FETCHING
});

const updateVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.UPDATE_VEHICLES_SUCCEEDED,
  vehicles
});

const updateVehiclesFailed = error => ({
  type: vehiclesAction.UPDATE_VEHICLES_FAILED,
  error
});

export const updateVehicles = page => dispatch => {
  dispatch(updateVehiclesIsFetching());
  return getVehicles(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updateVehiclesSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updateVehiclesFailed(error));
    });
};
