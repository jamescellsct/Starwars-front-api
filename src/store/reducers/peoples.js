import { peoplesAction } from "../actions/actionsTypes";

export default function peoples(state = null, action) {
  switch (action.type) {
    case peoplesAction.FETCH_PEOPLE_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case peoplesAction.FETCH_PEOPLE_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...(state.pages || []),
          { page: action.people.page, list: action.people.results }
        ],
        pageCount: action.people.pageCount,
        isFetching: false
      });

    case peoplesAction.FETCH_PEOPLE_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    case peoplesAction.UPDATE_PEOPLE_IS_FETCHING:
      return Object.assign({}, state, {
        isFetching: true
      });
    case peoplesAction.UPDATE_PEOPLE_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...state.pages,
          { page: action.people.page, list: action.people.results }
        ],
        pageCount: action.people.pageCount,
        isFetching: false
      });

    case peoplesAction.UPDATE_PEOPLE_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
