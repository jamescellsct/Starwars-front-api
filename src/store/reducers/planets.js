import { planetsAction } from "../actions/actionsTypes";

export default function planets(state = null, action) {
  switch (action.type) {
    case planetsAction.FETCH_PLANETS_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case planetsAction.FETCH_PLANETS_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...(state.pages || []),
          { page: action.planets.page, list: action.planets.results }
        ],
        pageCount: action.planets.pageCount,
        isFetching: false
      });

    case planetsAction.FETCH_PLANETS_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    case planetsAction.UPDATE_PLANETS_IS_FETCHING:
      return Object.assign({}, state, {
        isFetching: true
      });
    case planetsAction.UPDATE_PLANETS_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...state.pages,
          { page: action.planets.page, list: action.planets.results }
        ],
        pageCount: action.planets.pageCount,
        isFetching: false
      });

    case planetsAction.UPDATE_PLANETS_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
