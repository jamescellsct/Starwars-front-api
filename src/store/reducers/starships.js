import { starshipsAction } from "../actions/actionsTypes";

export default function starships(state = null, action) {
  switch (action.type) {
    case starshipsAction.FETCH_STARSHIPS_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case starshipsAction.FETCH_STARSHIPS_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...(state.pages || []),
          { page: action.starships.page, list: action.starships.results }
        ],
        pageCount: action.starships.pageCount,
        isFetching: false
      });

    case starshipsAction.FETCH_STARSHIPS_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    case starshipsAction.UPDATE_STARSHIPS_IS_FETCHING:
      return Object.assign({}, state, {
        isFetching: true
      });
    case starshipsAction.UPDATE_STARSHIPS_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...(state.pages || []),
          { page: action.starships.page, list: action.starships.results }
        ],
        pageCount: action.starships.pageCount,
        isFetching: false
      });

    case starshipsAction.UPDATE_STARSHIPS_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
