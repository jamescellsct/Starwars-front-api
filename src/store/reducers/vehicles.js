import { vehiclesAction } from "../actions/actionsTypes";

export default function vehicles(state = null, action) {
  switch (action.type) {
    case vehiclesAction.FETCH_VEHICLES_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case vehiclesAction.FETCH_VEHICLES_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...(state.pages || []),
          { page: action.vehicles.page, list: action.vehicles.results }
        ],
        pageCount: action.vehicles.pageCount,
        isFetching: false
      });

    case vehiclesAction.FETCH_VEHICLES_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    case vehiclesAction.UPDATE_VEHICLES_IS_FETCHING:
      return Object.assign({}, state, {
        isFetching: true
      });
    case vehiclesAction.UPDATE_VEHICLES_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...state.pages,
          { page: action.vehicles.page, list: action.vehicles.results }
        ],
        pageCount: action.vehicles.pageCount,
        isFetching: false
      });

    case vehiclesAction.UPDATE_VEHICLES_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
