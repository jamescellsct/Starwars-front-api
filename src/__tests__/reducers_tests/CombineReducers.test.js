import { createStore } from "redux";

import rootReducer from "../../store/reducers";

import pagination from "../../store/reducers/pagination";
import peoples from "../../store/reducers/peoples";
import planets from "../../store/reducers/planets";
import starships from "../../store/reducers/starships";
import vehicles from "../../store/reducers/vehicles";

const store = createStore(rootReducer);

describe("Testing combineReducers method in index.js from reducers", () => {
  it("Should render correctly pagination reducer", () => {
    expect(store.getState().pagination).toEqual(pagination({ page: 1 }, {}));
    expect(store.getState().pagination).toMatchSnapshot();
  });

  it("Should render correctly peoples reducer", () => {
    expect(store.getState().peoples).toEqual(peoples(null, {}));
    expect(store.getState().peoples).toMatchSnapshot();
  });

  it("Should render correctly planets reducer", () => {
    expect(store.getState().planets).toEqual(planets(null, {}));
    expect(store.getState().planets).toMatchSnapshot();
  });

  it("Should render correctly starships reducer", () => {
    expect(store.getState().starships).toEqual(starships(null, {}));
    expect(store.getState().starships).toMatchSnapshot();
  });

  it("Should render correctly vehicles reducer", () => {
    expect(store.getState().vehicles).toEqual(vehicles(null, {}));
    expect(store.getState().vehicles).toMatchSnapshot();
  });
});
