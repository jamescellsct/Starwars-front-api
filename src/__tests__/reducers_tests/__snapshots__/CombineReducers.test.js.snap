// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing combineReducers method in index.js from reducers Should render correctly pagination reducer 1`] = `
Object {
  "page": 1,
}
`;

exports[`Testing combineReducers method in index.js from reducers Should render correctly peoples reducer 1`] = `null`;

exports[`Testing combineReducers method in index.js from reducers Should render correctly planets reducer 1`] = `null`;

exports[`Testing combineReducers method in index.js from reducers Should render correctly starships reducer 1`] = `null`;

exports[`Testing combineReducers method in index.js from reducers Should render correctly vehicles reducer 1`] = `null`;
